src/jit/block.cpp

```diff
@@ -1374,6 +1374,56 @@ BasicBlock* Compiler::bbNewBasicBlock(BBjumpKinds jumpKind)
     return block;
 }
 
+//------------------------------------------------------------------------
+// blockhasEHBoundaryIn: Determine if this block begins at an EH boundary.
+//
+// Return Value:
+//    True iff the block is the target of an EH edge; false otherwise.
+//
+bool BasicBlock::hasEHBoundaryIn()
+{
+#if FEATURE_EH_FUNCLETS
+    return ((bbCatchTyp != BBCT_NONE) || (bbFlags & BBF_FUNCLET_BEG));
+#else  // !FEATURE_EH_FUNCLETS
+    return (bbCatchTyp != BBCT_NONE);
```

Comment:
It _seems_ like just checking `BBCT_NONE` would be sufficient, even in the funclets case, since IIRC, `BBF_FUNCLET_BEG` being set will imply `bbCatchTyp != BBCT_NONE`.